backend/capellacollab/projects/toolmodels/diagrams/routes.py

```diff
@@ -67,14 +68,24 @@ async def get_diagram_metadata(
     )
 
 
-@router.get("/{diagram_uuid}", response_class=fastapi.responses.Response)
+@router.get(
+    "/{diagram_uuid_or_filename}", response_class=fastapi.responses.Response
+)
 async def get_diagram(
-    diagram_uuid: str,
+    diagram_uuid_or_filename: str,
     handler: git_handler.GitHandler = fastapi.Depends(
         git_injectables.get_git_handler
     ),
     logger: logging.LoggerAdapter = fastapi.Depends(log.get_request_logger),
 ):
+    fileextension = pathlib.PurePosixPath(diagram_uuid_or_filename).suffix
+    if fileextension and fileextension.lower() != ".svg":
+        raise fastapi.HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
```

Comment:
Should be a bad request, as we don't support SVG and the user requested SVG.

Reply:
I think @Wuestengecko said that in case capellembse requests another file format it only understands if exactly a 404 is thrown...

Reply:
Oh, alright. I forgot to reply to that, sorry.

In my opinion, this is exactly what should happen. The user tries to use a feature that we don't support. In this case, it's a wrong configuration - the diagram cache doesn't work with the configuration. When I as user pass the configuration that I want to use the diagram cache, then I'd also expect that it either works or an error is raised, it should not be ignored silently.

Reply:
alright - so I will change it to 400 Bad request